### PR TITLE
Add crd files and example

### PR DESCRIPTION
**Pull request by xiaolanz**
A minimum broker CRD resource definition. #15

### CODE REVIEW

**Comment by xiaolanz** [comment]
PTAL the yaml definitions. 
tests should be fixed after I merge #18

**Comment by ayj** [comment]
Looks like the content under `spec` is using definitions from `//pkg/model`. These might need to eventually be defined in `istio.io/api/broker/v1/config` similar to Mixer and Proxy.

**Comment by xiaolanz** [comment]
the one under `//pkg/model` is the OSBI structure. It's included in but different from `spec` because `spec` contains istio internal information like service. I will add a config in api for the spec.

File: example/demo_catalog.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+apiVersion: "config.istio.io/v1alpha2"
```
Comment (xiaolanz) [inline example/demo_catalog.yaml]: should I use v1alpha2 or v1alpha1? @kyessenov

Comment (kyessenov) [inline example/demo_catalog.yaml]: v1alpha2 or v2beta1. still debating...

**Comment by xiaolanz** [comment]
I've put up a PR for api/broker https://github.com/istio/api/pull/156

File: example/demo_catalog.yaml (hunk 1 above)
Comment (ayj) [inline example/demo_catalog.yaml]: [istio/api](https://github.com/istio/api) configuration is versioned per component. Seems like the same versioning should be reflected in the CRD types, e.g.

```yaml
apiversion: broker.config.istio.io/v1alpha2   # => istio/api/broker/v1
apiVersion: proxy.config.istio.io/v1beta1     # => istio/api/proxy/v1
apiversion: mixer.config.istio.io/v1beta2     # => istio/api/mixer/v1
```

Comment (jmuk) [inline example/demo_catalog.yaml]: If we separate the api group by istio component, maybe we can skip `config`?  Like `broker.istio.io/v1alpha2`

Comment (xiaolanz) [reply example/demo_catalog.yaml]: just v1 doesn't work. I will keep to use v1alpha2

**Pull request merged**